pkgs/development/haskell-modules/configuration-common.nix

```
@@ -1402,6 +1402,17 @@ self: super: {
       '';
     }));
 
+  # Runtime dependencies and CLI completion
+  nvfetcher = generateOptparseApplicativeCompletion "nvfetcher" (overrideCabal
+    super.nvfetcher (drv: {
+      buildTools = drv.buildTools or [ ] ++ [ pkgs.makeWrapper ];
+      postInstall = drv.postInstall or "" + ''
+        wrapProgram "$out/bin/nvfetcher" --prefix 'PATH' ':' "${
+          pkgs.lib.makeBinPath [ pkgs.nvchecker pkgs.nix-prefetch-git ]
```

This ensures the derivation refers to *itself* in `PATH` rather than to `pkgs.nvfetcher` which would make `pkgs.haskellPackages.nvfetcher` depend on `pkgs.nvfetcher` which would be quite unnecessary.

```suggestion
          pkgs.lib.makeBinPath [ (placeholder "out") pkgs.nix-prefetch-git ]
```

---

It needs 'pkg.nvchecker', rather than itself.

---

Ah, I was confused by this as well.

@sternenseemann Note that it is `nvchecker`, not `nvfetcher`.